custom_components/dualmode_generic/climate.py

```diff
@@ -330,17 +347,25 @@ async def async_set_hvac_mode(self, hvac_mode):
             self._hvac_mode = HVAC_MODE_HEAT
             if self._is_device_active and not self.reverse_cycle:
                 await self._async_cooler_turn_off()
+                await self._async_fan_turn_off()
             await self._async_control_heating(force=True)
         elif hvac_mode == HVAC_MODE_COOL:
             self._hvac_mode = HVAC_MODE_COOL
             if self._is_device_active and not self.reverse_cycle:
                 await self._async_heater_turn_off()
+                await self._async_fan_turn_off()
             await self._async_control_heating(force=True)
+        elif hvac_mode == HVAC_MODE_FAN_ONLY:
+            self._hvac_mode = HVAC_MODE_FAN_ONLY
+            if self._is_device_active:
```

Comment:
With the `reverse_cycle` mode turned on (cooler and heater the same device), "fan mode" will need to respect this (see line 354) so turning the heater or cooler off won't also turn the fan off. 
Here in lies a problem, though - what if the cooler and ventilator/fan ARE the same device, but not the heater. Or, what if the heater and the fan are the same device, but not the cooler? Or, what if the heater and the cooler are the same device, but not the ventilator? This function is going to start to get a bit complicated....

Reply:
Disclaimer: I'm using an IR Wifi Controller to send signals and a mobile HVAC

I admit I don't fully understand how reverse_cycle works...
Nevertheless, I thought that you have to provide a cooler and heater as separate entities. Given that, I set up my stuff with input_booleans and then added automation listening for their states. So when switching modes, the mobile HVAC is first turned off by "heater" and then turned on again by "fan" and then gets the control corresponding signals.

Reply:
It sounds like the options are:
1. Require that the heater, cooler, and fan/ventilator entities are unique. This would break the existing `reverse_cycle` implementation. 
2. Have a complex conditional in this function for every possible permutation of heater/cooler/fan and checking whether they are the same entities. 

If those are the options, I think my preference would be for the first, primarily because `reverse_cycle` is more of an edge case than a `fan` is. We would then have to ask anyone with a `reverse_cycle` device to obfuscate that behind some switch templates.

Reply:
well, actually there's a third option...
For those that don't mind not being able to use the fan property/entity, they could just keep using the integration as is ;D

Reply:
> Nevertheless, I thought that you have to provide a cooler and heater as separate entities. Given that, I set up my stuff with input_booleans and then added automation listening for their states. 

Ah right. The `reverse_cycle` mode allows you to avoid having to set up these control input booleans and associated automations (sounds similar to the hack I had before I got the 💩 and created the PR) - does this by explicitly NOT turning off the other switch when turning on, to prevent the device from switching off while it should be switching on.

Reply:
> If those are the options, I think my preference would be for the first, primarily because `reverse_cycle` is more of an edge case than a `fan` is. 

I dunno - here in Australia, reverse cycle air conditioners are everywhere - especially in milder zones (like where I am in Sydney). I only have a couple of plug-in heaters that I've kept around for the really cold nights (rare) and edges of the house that the air-con heater doesn't quite reach.

To still support reverse_cycle for this new mode, you would just need to change the way the reverse_cycle mode is configured. Currently it's just a simple bool flag, but it could be configured with list of the modes that involve the same device - e.g.:
`reverse_cycle: [COOL, HEAT] # for saying cooler and heater are the same device`
Then use that value to work out what to/not to switch off when setting a mode.